nav2_behavior_tree/include/nav2_behavior_tree/plugins/condition/generic_error_code_condition.hpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2022 Joshua Wallace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__CONDITION__GENERIC_ERROR_CODE_CONDITION_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__CONDITION__GENERIC_ERROR_CODE_CONDITION_HPP_
+
+#include <string>
+#include <memory>
+#include <vector>
+#include <set>
+
+#include "rclcpp/rclcpp.hpp"
+#include "behaviortree_cpp_v3/condition_node.h"
+
+namespace nav2_behavior_tree
+{
+
+class GenericErrorCodeCondition : public BT::ConditionNode
```

Comment:
I think something semantically more meaningful for a name in the BT could help. Something in the theme of the verbal-naming conventions we use for the other condition nodes would be good. That way reading the name you undersatnd what SUCCESS or FAILURE mean. `GenericErrorCodeCondition` doesn't tell me what true or false results indicate

Reply:
How about `IsErrorCodeActive`?

Reply:
I like that one